starter_code/elevator.js

```diff
@@ -1,19 +1,75 @@
+'use strict';
+const people = require('./person');
+
 class Elevator {
-  constructor(){
-    this.floor      = 0;
-    this.MAXFLOOR   = 10;
-    this.requests   = [];
-  }
-
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  constructor() {
+    this.floor = 0;
+    this.MAXFLOOR = 10;
+    this.requests = [];
+    this.waitingList = [];
+    this.passengers = [];
+  }
+
+  start() {
+    this.startInterval = setInterval(update(), 1000);
+  }
+
+  stop() {
+    clearInterval(this.startInterval);
+  }
+
+  update() {
+    this.log();
+  }
+
+  passengersEnter() {
+    if (this.waitingList.length > 0) {
+      this.passengers.push(people.Person.name);
+      this.waitingList.shift();
+      this.requests.push(people.Person.destinationFloor);
+      console.log(people.Person.name + ' has enter the elevator');
+    }
+  }
+
+  passengersLeave() {}
+
+  floorUp() {
+    if (this.floor < 10) {
+      this.floor += 1;
+    }
+    console.log(this.floor);
+  }
+
+  floorDown() {
+    if (this.floor > 0) {
+      this.floor -= 1;
+    }
+    console.log(this.floor);
+  }
+
+  call() {
+    this.waitingList.push(people.Person);
+    this.requests.push(people.Person.originFloor);
+  }
+
+  log() {
+    if (this.floorUp) {
```

Comment:
Two IMPORTANT problems:
1. this.floorUp doesn't return a boolean, bad choice putting it inside an if.
2. This line calls the method this.floorUp, so you move the elevator up everytime you call this.log